tests/testthat/test-examples.R

```diff
@@ -0,0 +1,118 @@
+# this test requires a `man` directory in the `tests/testthat` directory
+# (presumably symlinked to the package root `man` directory to avoid duplication)
+# this also requires `devtools::document()` to be run before running the tests
+
+rd_files <- function() {
+  list.files(testthat::test_path("man"), pattern = "\\.[Rr]d$", full.names = TRUE)
+}
+
+suppress_warnings <- function(expr, pattern = "*", ...) {
+  withCallingHandlers(
+    expr,
+    warning = function(w) {
+      if (grepl(pattern, conditionMessage(w))) {
+        invokeRestart("muffleWarning")
+      }
+    }
+  )
+}
+
+with_mocked_app_bindings <- function(code) {
+  shiny__shinyApp <- shiny::shinyApp # nolint object_name_linter.
+
+  # workaround of https://github.com/rstudio/shinytest2/issues/381
+  # change to `print(shiny__shinyApp(...))` once fixed
+  mocked_shinyApp <- function(ui, server, ...) { # nolint object_name_linter.
+    functionBody(server) <- bquote({
+      library(.(testthat::testing_package()), character.only = TRUE)
+      .(functionBody(server))
+    })
+    print(do.call(shiny__shinyApp, append(x = list(ui = ui, server = server), list(...))))
+  }
+
+  mocked_runApp <- function(x, ...) { # nolint object_name_linter.
+    app_driver <- shinytest2::AppDriver$new(
+      x,
+      shiny_args = list(...),
```

Comment:
I experienced some timeouts the first time I ran the tests. (namely the `load_timeout` of 15s)

I suggest that we replicate here the default options we use for `TealAppDriver`. Consequently, the `app_driver$wait_for_idle` no longer needs the parameter.

```suggestion
      shiny_args = list(...),
      timeout = 20000,
      load_timeout = 100000,
```

Reply:
`load_timeout = 100000`
that means 100 secs per each app. This is a lot

Reply:
Yes, that's overkill :sweat_smile: 

However, keeping at 15s may cause intermittent problems depending on the machine/VM where it's being tested.

Reply:
I'm merging this as is. If it's not enough please open a separate PR!